apps/studio/src/features/editing-experience/components/TipTapComponent.tsx

```diff
@@ -20,55 +21,65 @@ import { Text } from '@tiptap/extension-text'
 import TableCell from '@tiptap/extension-table-cell'
 import TableHeader from '@tiptap/extension-table-header'
 import TableRow from '@tiptap/extension-table-row'
-import { EditorContent, useEditor } from '@tiptap/react'
-// import type { CustomRendererProps } from './types'
-import { BiImage, BiText, BiX } from 'react-icons/bi'
+import {
+  type Content,
+  EditorContent,
+  useEditor,
+  type JSONContent,
+} from '@tiptap/react'
+import { BiText, BiX } from 'react-icons/bi'
 import Underline from '@tiptap/extension-underline'
 import { MenuBar } from '~/components/PageEditor/MenuBar'
 
-import { Table } from './extensions/Table'
 import { useEditorDrawerContext } from '~/contexts/EditorDrawerContext'
-
-type NativeComponentType = 'paragraph' | 'image'
+import { type IsomerNativeComponentProps } from '@opengovsg/isomer-components'
+import { cloneDeep } from 'lodash'
+import { Table } from './extensions/Table'
+import { validateAsProse } from '../utils/convert'
 
 export interface TipTapComponentProps {
-  type: NativeComponentType
-  data: any
-  path: string
+  type: IsomerNativeComponentProps['type']
 }
 
 const typeMapping = {
-  paragraph: {
+  prose: {
     icon: BiText,
-    title: 'Paragraph',
-  },
-  image: {
-    icon: BiImage,
-    title: 'Image',
+    title: 'Prose',
   },
 }
 
-function TipTapComponent({ type, data, path }: TipTapComponentProps) {
-  const { setDrawerState, setPageState, setEditorState } =
+function TipTapComponent({ type }: TipTapComponentProps) {
+  const { setDrawerState, setPageState, currActiveIdx, editorState } =
     useEditorDrawerContext()
+
+  const updatePageState = (editorContent: JSONContent) => {
+    const content = validateAsProse(editorContent)
+    setPageState((oldState) => {
+      // TODO: performance - this is a full clone
+      // of the object, which is expensive
+      const newState = cloneDeep(oldState)
+      newState[currActiveIdx] = content
+      return newState
+    })
+  }
+
   const editor = useEditor({
     extensions: [
       Blockquote,
       Bold,
       BulletList.extend({
-        name: 'unorderedlist',
-      }),
-      BulletList.configure({
+        name: 'unorderedList',
+      }).configure({
         HTMLAttributes: {
           class: 'list-disc',
         },
       }),
```

Comment:
this is a bug fix, previously this returned 2 instances of `BulletList` (1 configured, 1 extended) and the later one won.